crates/oxc_codegen/src/lib.rs

```diff
@@ -61,6 +69,16 @@ pub struct Codegen<const MINIFY: bool> {
 
     /// Track the current indentation level
     indentation: u8,
+
+    // sourcemap
+    enable_sourcemap: bool,
```

Comment:
It would be nice to have a `SourceMapBuilder` for encapsulating all these states and methods.